### ISSUE TITLE
Audit diagnostic messages for missing quotes around symbol names

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn-analyzers#1714]
I think quite a few of our diagnostic messages were created by an automated tool that ported messages from old FxCop implementation. For example, see https://github.com/dotnet/roslyn-analyzers/issues/1713#issuecomment-395051433.

**Comment by Evangelink** [comment]
@mavasani Do you have some guidelines around when the "thing" should be single quoted? Is it like every time this is a keyword or a type/method/property/... name?

**Comment by Evangelink** [comment]
ping @mavasani

**Comment by mavasani** [comment]
Yes - I think at the very least all symbol names in messages should be within single quotes for clarity.

**Comment by Evangelink** [comment]
To sum-up some of the exchanges on the 1st draft PR:

- Prefer `SomeAttribute` for diagnostics, and `[Some]` for code fixes.

- Prefer no '.' for single-sentence titles and messages (and avoid multi-sentence unless it substantially detracts from clarity)

- Use full sentences with '.' for descriptions

- Do not use pascal case for all words
